workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/SpatialOSConnectionHandler.cs

```diff
@@ -0,0 +1,52 @@
+using Improbable.Worker.CInterop;
+
+namespace Improbable.Gdk.Core
+{
+    public class SpatialOSConnectionHandler : IConnectionHandler
+    {
+        private readonly ViewDiff diff = new ViewDiff();
+        private readonly OpListDeserializer deserializer = new OpListDeserializer();
+        private readonly SerializedMessagesToSend messagesToSend = new SerializedMessagesToSend();
+
+        private readonly CommandMetaData commandMetaData = new CommandMetaData();
+
+        private readonly Connection connection;
+
+        public SpatialOSConnectionHandler(Connection connection)
+        {
+            this.connection = connection;
+        }
+
+        public bool IsConnected()
+        {
+            return connection.GetConnectionStatusCode() == ConnectionStatusCode.Success;
+        }
+
+        public ViewDiff GetMessagesReceived()
+        {
+            commandMetaData.FlushRemovedIds();
+            diff.Clear();
+
+            bool inCriticalSection = false;
+            do
+            {
+                using (var opList = connection.GetOpList(0))
+                {
+                    inCriticalSection = deserializer.ParseOpListIntoDiff(opList, diff, commandMetaData);
+                }
+            }
+            while (inCriticalSection);
+
+            deserializer.Reset();
+
+            return diff;
+        }
+
+        public void PushMessagesToSend(MessagesToSend messages)
```

Comment:
Bit of bikeshedding - not a huge fan of this name, can't think of anything else right now

Reply:
This can be changed any time. Will leave it like this for the moment, but it can be added to the naming tasks.